src/Entities/Misc/DreamStrawberry.cs

```diff
@@ -0,0 +1,135 @@
+
+            dreamStrawberrySpriteData = DynamicData.For(this);
+        }
+
+        #region Bad Code
+        /*
+        public override void Added(Scene scene) {
+            sprite.OnFrameChange = OnAnimate;
+            Add(wiggler = Wiggler.Create(0.4f, 4f, delegate (float v)
+            {
+                sprite.Scale = Vector2.One * (1f + v * 0.35f);
+            }));
+            Add(rotateWiggler = Wiggler.Create(0.5f, 4f, delegate (float v)
+            {
+                sprite.Rotation = v * 30f * ((float) Math.PI / 180f);
+            }));
+            Add(bloom = new BloomPoint(1f, 12f));
+            Add(light = new VertexLight(Color.White, 1f, 16, 24));
+            Add(lightTween = light.CreatePulseTween());
+            if ((scene as Level).Session.BloomBaseAdd > 0.1f) {
+                bloom.Alpha *= 0.5f;
+            }
+        }
+
+        public void OnAnimate(string id) {
+            if (!flyingAway && id == "flap" && sprite.CurrentAnimationFrame % 9 == 4) {
+                Audio.Play("event:/game/general/strawberry_wingflap", Position);
+                flapSpeed = -50f;
+            }
+            int num = 25;
+            // Checks if the animation from is 25, runs the funny strawberry noises
+            if (sprite.CurrentAnimationFrame == num) {
+                lightTween.Start();
+                if (!collected && (CollideCheck<FakeWall>() || CollideCheck<Solid>())) {
+                    Audio.Play("event:/game/general/strawberry_pulse", Position);
+                    SceneAs<Level>().Displacement.AddBurst(Position, 0.6f, 4f, 28f, 0.1f);
+                } else {
+                    Audio.Play("event:/game/general/strawberry_pulse", Position);
+                    SceneAs<Level>().Displacement.AddBurst(Position, 0.6f, 4f, 28f, 0.2f);
+                }
+            }
+        }
+        */
+        #endregion
+
+        public override void Update() {
+
+            base.Update();
+
+            // Creates and updates the dream trail
+            if (Scene.OnInterval(0.1f))
+                CreateDreamTrail(this);
+        }
+
+        // Calls the original OnDash from our DreamDashListener
+        public void OnDreamBerryDash(Vector2 dir) {
+            OriginalOnDash.Invoke(this, new object[] { dir });
+        }
+
+        // Code to create a trail for the berry to make it separate from normal berries
+        public static void CreateTrail(DreamStrawberry berry, Color color) {
+            Sprite berrySprite = dreamStrawberrySpriteData.Get<Sprite>("sprite");
+            Vector2 scale = new Vector2(Math.Abs(berrySprite.Scale.X), berrySprite.Scale.Y);
+            TrailManager.Add(berry, scale, color);
+        }
+
+        // Actually creates the dream trail
+        public static void CreateDreamTrail(DreamStrawberry berry) {
+            CreateTrail(berry, DreamTrailColors[DreamTrailColorIndex]);
+            ++DreamTrailColorIndex;
+            DreamTrailColorIndex %= 5;
+        }
```

Comment:
These methods can probably be merged, and don't have to be marked `static`. Instead of passing the `DreamStrawberry` object as a parameter, don't make it static, and use directly reference the things you need from the instance (or with `this`), because that method now belongs to the `DreamStrawberry` instance.